src/main/java/net/dv8tion/jda/api/entities/User.java

```diff
@@ -220,6 +223,50 @@ default String getEffectiveAvatarUrl()
         return avatarUrl == null ? getDefaultAvatarUrl() : avatarUrl;
     }
 
+    /**
+     * The Discord Id for this user's banner image.
+     * If the user has not set a banner, this will return null.
+     * If the user has not been retrieved using {@link JDA#retrieveUserById(long)}, this will return null.
```

Comment:
Does this mean you don't get banner details via gateway? Maybe we should make this be like Guild#retrieveMetaData then? How does this affect events?

Reply:
Yes, `banner` and `accent_color` are only sent when the user is fetched via `GET /users/:user_id`. The event. fires when the user is retrieved but not when the change actually occurs.

Reply:
How about `User#retrieveProfile` which returns a `Profile` POJO that contains all this REST-only data?

Reply:
Should this replace the existing getters or exist alongside them (like with guild metadata)?

Reply:
If we can't keep this information up to date via the gateway I don't enjoy the idea of providing getters like this.
Makes me think we should scrap any top-level getters and go exclusively for the `retrieveProfile` _only_.

Reply:
The problem with that is we lose the information from `retrieveUserById`

Reply:
We really need a SocketUser and RestUser huh..